src/biogeochem/DustEmisLeung2023.F90

```diff
@@ -0,0 +1,874 @@
+         tmp2 = sqrt(A_Shao * (dns_slt*grav*D_p + gamma_Shao/D_p))  ! calculate S&L (2000) scheme here for threshold; gamma = 1.65e-4 following S&L00, D_p = 127 um ~ 130 um following Leung et al. (2023). dmleung use defined parameters instead of typing numerical values 16 Feb 2024
+         wnd_frc_thr_dry(p) = tmp2 / sqrt(forc_rho(c))    ! dry fluid threshold
+         wnd_frc_thr_slt = tmp2 / sqrt(forc_rho(c)) * frc_thr_wet_fct !* frc_thr_rgh_fct   ! fluid threshold. dmleung commented out frc_thr_rgh_fct since it is used to modify the wind, not the wind threshold.
+         wnd_frc_thr_slt_it = B_it * tmp2 / sqrt(forc_rho(c)) ! define impact threshold
+
+         ! the above formula is true for Iversen and White (1982) and Shao and Lu (2000) scheme
+         wnd_frc_thr(p) = wnd_frc_thr_slt          ! output fluid threshold
+         wnd_frc_thr_it(p) = wnd_frc_thr_slt_it    ! output impact threshold
+
+         !##############################################################################################
+         ! dmleung: here, calculate quantities relevant to the fluid threshold
+         ! standardized fluid threshold
+         wnd_frc_thr_slt_std = wnd_frc_thr_slt * sqrt(forc_rho(c) / forc_rho_std) ! standardized soil threshold friction speed (defined using fluid threshold)
+         wnd_frc_thr_std(p) = wnd_frc_thr_slt_std          ! output standardized fluid threshold
+         ! dust emission coefficient or soil erodibility coefficient (this is analogous to the soil erodibility map or prefenertial source filter in Zender; see zendersoilerodstream)
+         dst_emiss_coeff(p) = Cd0 * exp(-Ce * (wnd_frc_thr_slt_std - wnd_frc_thr_slt_std_min) / wnd_frc_thr_slt_std_min) ! save dust emission coefficient here for all grids
+
+         ! framentation exponent (dependent on fluid threshold)
+         frag_expt = (Ca * (wnd_frc_thr_slt_std - wnd_frc_thr_slt_std_min) / wnd_frc_thr_slt_std_min)  ! fragmentation exponent, defined in Kok et al. (2014a)
+         if (frag_expt > frag_expt_thr) then   ! set fragmentation exponent to be 3 or 5 at maximum, to avoid local AOD blowup
+            frag_expt = frag_expt_thr
+         end if
+
+         !##############################################################################################
+         !################ drag partition effect, and soil-surface friction velocity ###################
+         ! subsection on computing vegetation drag partition and hybrid drag partition factors
+         ! in Leung et al. (2023), drag partition effect is applied on the wind instead of the threshold
+         !##############################################################################################
+         ! the following comes from subr. frc_thr_rgh_fct_get
+         ! purpose: compute factor by which surface roughness increases threshold
+         !          friction velocity (currently a constant)
+
+         if (lnd_frc_mbl(p) > 0.0_r8  .AND. tlai_lu(l)<=1_r8) then
```

Comment:
@dmleung should the 1._r8 here really be vai_mbl_thr?

There's a couple other places below that I thought might also be true below.